### ISSUE TITLE
The back tooltip floats when we return from the Search.

### ISSUE THREAD

**Issue by m-natarajan** [Expensify/App#36053]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.4.38-0
**Reproducible in staging?:** y
**Reproducible in production?:** n
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** Applause internal team
**Slack conversation:**

## Action Performed:
1, Click on "Search."
2, Upon returning from the search, observe that the back tooltip unexpectedly floats.
## Expected Result:
The back tooltip should disappear when the modal is closed.
## Actual Result:
The back tooltip floats when we return from the Search.

## Workaround:
Can the user still use Expensify without this being fixed? Have you informed them of the workaround?

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

https://github.com/Expensify/App/assets/38435837/cb6366fd-a889-4b67-af35-3c7b3b31a783

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by cead22** [comment]
This doesn't feel like a blocker, @greg-schroeder can you put it through the regular bug process please?

**Comment by greg-schroeder** [comment]
Agreed not a blocker, setting `External` and leaving `Daily`

**Comment by greg-schroeder** [comment]
Yes I did, that was an accident. I've rescinded that offer, silly automatation 😅

**Comment by nduisekeyev** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.

The back tooltip floats from the workspace switcher.

### What is the root cause of that problem?

The absence of the second argument in the `HeaderWithBackButton` from the `Navigation` library, which is responsible for enforcing navigation to the fallback route, results in the floating of the back tooltip.

https://github.com/Expensify/App/blob/9922a399079df0fbf820673b351991fddf97fdf3/src/libs/Navigation/Navigation.ts#L185

### What changes do you think we should make in order to solve the problem?

By passing `Navigation.goBack(ROUTES.HOME, true)` to the `onBackButtonPress` prop, we're ensuring that the `goBack` function is called with the appropriate arguments, including the fallback route `(ROUTES.HOME)` and the `shouldEnforceFallback` flag set to true in the `HeaderWithBackButton`.

https://github.com/Expensify/App/blob/9922a399079df0fbf820673b351991fddf97fdf3/src/components/HeaderWithBackButton/index.tsx#L29


### Screenshots/Videos

https://github.com/Expensify/App/assets/26330812/933a03e3-a31c-467a-9680-25a59fb45786



### What alternative solutions did you explore? (Optional)
N/A

**Comment by nduisekeyev** [comment]
Contributor details
Your Expensify account email: nur.duisekeyev@hotmail.com
Upwork Profile Link: https://www.upwork.com/fl/~01ab96855ba9fc9fc4

**Comment by greg-schroeder** [comment]
Adding retest label to have QA confirm if this is still an issue

**Comment by greg-schroeder** [comment]
Waiting for reproduction

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (First week)

**Comment by greg-schroeder** [comment]
Closing this after multiple failures to reproduce, reopen if you disagree
